src/client/src/routes/SimpleLauncher/LaunchButton.tsx

```diff
@@ -45,10 +45,9 @@ const StyledButton = styled.button`
 `
 
 export class LaunchButton extends React.Component<LaunchButtonProps> {
-  onClick = (event: React.MouseEvent<HTMLButtonElement>) => {
-    open(this.props.appConfig)
-    event.preventDefault()
-  }
+  onClick = (event: React.MouseEvent<HTMLButtonElement>) => (
+    this.props.appConfig.launch ? this.props.appConfig.launch() : open(this.props.appConfig), event.preventDefault()
```

Comment:
This component does not need to know about the app config. In general, ask yourself
1. Does this files/class do one thing only and one thing well
2. Does this file/class know more than it needs? For example, an open function instead of the app config